### PR TITLE
feat: add didcomm V2 WACI share support

### PR DESCRIPTION
**Pull request by Baha-sk**
Signed-off-by: Baha Shaaban <baha.shaaban@securekey.com>

### CODE REVIEW

File: test/fixtures/wallet-web/docker-compose-demo.yml [hunk 1]
```diff
@@ -17,8 +17,12 @@ services:
       - TLS_CACERTS=/etc/tls/ec-cacert.pem
       - TLS_KEY_FILE=/etc/tls/ec-key.pem
       - TLS_CERT_FILE=/etc/tls/ec-pubCert.pem
-      - ORB_DOMAIN=testnet.orb.local
+      - ORB_DOMAIN=https://testnet.orb.local
+      - ROUTER_DOMAIN=https://hub-router.trustbloc.local:${HUB_ROUTER_HTTP_INBOUND_PORT}
       - CONTEXT_PROVIDER_URL=${CONTEXT_PROVIDER_URL}
+      - USE_DIDCOMM_V2=${MOCK_ADAPTER_USE_DIDCOMM_V2}
```
Comment (fqutishat) [inline test/fixtures/wallet-web/docker-compose-demo.yml:23]: Default should be DIDcomm v2

Comment (Baha-sk) [reply test/fixtures/wallet-web/docker-compose-demo.yml:23]: yes, it's in the env file above `=true`

Comment (fqutishat) [reply test/fixtures/wallet-web/docker-compose-demo.yml:23]: @Baha-sk I mean in same app

**Review by fqutishat — APPROVED** [review]

File: cmd/wallet-web/src/mixins/common/register.js [hunk 2]
```diff
@@ -33,7 +33,7 @@ export class RegisterWallet {
     let failure;
     try {
       await Promise.all([
-        this._connectToMediator(),
+        this._connectToMediator(true),
```
Comment (sudeshrshetty) [inline cmd/wallet-web/src/mixins/common/register.js:36]: has to be read from profile, can not hardcode

File: cmd/wallet-web/src/store/modules/options.js [hunk 3]
```diff
@@ -157,6 +169,14 @@ export default {
         agentOpts.walletMediatorURL = 'https://localhost:10093';
         agentOpts.hubAuthURL = 'https://localhost:8044';
 
+        let mediaTypeProfiles = ['didcomm/v2'];
```
Comment (sudeshrshetty) [inline cmd/wallet-web/src/store/modules/options.js:172]: revert this change you did for dev mode, you are not using it

File: images/wallet-web/templates/default.conf.template [hunk 4]
```diff
@@ -84,7 +84,10 @@ server {
             "sidetreeToken":"${SIDETREE_TOKEN}",
             "hubAuthURL":"${HTTP_SERVER_HUB_AUTH_URL}",
             "staticAssetsUrl": "${STATIC_ASSETS_URL}",
-            "context-provider-url":"${CONTEXT_PROVIDER_URL}"
+            "context-provider-url":"${CONTEXT_PROVIDER_URL}",
+            "media-type-profiles": "p256kw",
```
Comment (sudeshrshetty) [inline images/wallet-web/templates/default.conf.template:88]: you cannot hardcode configurations here

File: test/fixtures/wallet-web/.env [hunk 5]
```diff
@@ -29,6 +29,10 @@ HUB_ROUTER_API_PORT=10093
 HUB_ROUTER_DB_PATH=/tmp/db/aries1
 HUB_ROUTER_WEBHOOK_PORT=10094
 HUB_ROUTER_AUTOACCEPT=true
+HUB_ROUTER_USE_DIDCOMM_V2=true
+HUB_ROUTER_ORB_DOMAINS=https://testnet.orb.local
+HUB_ROUTER_V2_KEY_TYPE=ecdsap256ieee1363
+HUB_ROUTER_V2_KEY_AGREEMENT_TYPE=p256kw
```
Comment (sudeshrshetty) [inline test/fixtures/wallet-web/.env:35]: Are you sure this config works for V1?

File: test/fixtures/wallet-web/docker-compose-demo.yml [hunk 6]
```diff
@@ -17,8 +17,12 @@ services:
       - TLS_CACERTS=/etc/tls/ec-cacert.pem
       - TLS_KEY_FILE=/etc/tls/ec-key.pem
       - TLS_CERT_FILE=/etc/tls/ec-pubCert.pem
-      - ORB_DOMAIN=testnet.orb.local
+      - ORB_DOMAIN=https://testnet.orb.local
+      - ROUTER_DOMAIN=https://hub-router.trustbloc.local:${HUB_ROUTER_HTTP_INBOUND_PORT}
       - CONTEXT_PROVIDER_URL=${CONTEXT_PROVIDER_URL}
+      - USE_DIDCOMM_V2=true
```
Comment (sudeshrshetty) [inline test/fixtures/wallet-web/docker-compose-demo.yml:23]: As discussed, create a follow up task for V1 tests/containers.

File: test/fixtures/wallet-web/docker-compose-web.yml [hunk 7]
```diff
@@ -26,7 +26,7 @@ services:
       - TLS_KEY=/etc/nginx/certs/ec-key.pem
       - EDV_SERVER_URL=https://edv-oathkeeper-proxy.trustbloc.local:4455/encrypted-data-vaults
       - STORAGE_TYPE=edv
-      - KMS_TYPE=webkms
+      - KMS_TYPE=local
```
Comment (sudeshrshetty) [inline test/fixtures/wallet-web/docker-compose-web.yml:29]: add a TODO to revert it (need a follow up task created)

**Review by sudeshrshetty — CHANGES_REQUESTED** [review]

File: test/fixtures/wallet-web/.env (hunk 5 above)
Comment (fqutishat) [reply test/fixtures/wallet-web/.env:35]: @sudeshrshetty it's ok for now. We can fix it when we test DIDcomm v1
